### PR TITLE
Add resolveConfig method and corresponding tests

### PR DESCRIPTION
**Pull request by szeck87**
I also added `paths` into `tsconfig.json` because the imports were getting a bit out of hand with `../` repeating.  Thanks to this article and author: https://levelup.gitconnected.com/are-we-there-yet-or-how-to-improve-typescript-sass-imports-bce09ecea164

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -27,15 +27,29 @@ export const beautifier: Beautifier = {
       parseVersion: [/version (.*) by/, /PHP CS Fixer (\d+\.\d+\.\d+)/],
     },
   ],
-  beautify({ text, dependencies, filePath }: BeautifierBeautifyData) {
+  resolveConfig: ({ filePath, projectPath }) => {
+    const configFiles: string[] = [".php_cs", ".php_cs.dist"];
+    return findFile({ finishPath: projectPath, startPath: filePath, fileNames: configFiles }).then(configFile => {
+      return Promise.resolve({ filePath: configFile });
```
Comment (Glavin001) [inline src/index.ts]:
```
.then(configFile => {
      return Promise.resolve({ filePath: configFile });
})
```

can simply be

```
.then(configFile => ({ filePath: configFile }))
```

File: src/index.ts [hunk 2]
```diff
@@ -53,6 +67,43 @@ export const beautifier: Beautifier = {
...
+  // Recursive case: Check & go uuuuppp
+  const filePaths = fileNames.map(fileName => path.join(startPath, fileName));
+  // tslint:disable-next-line
+  return Promise.all(filePaths.map(filePath => doesFileExist(filePath)))
+  .then(exists => filePaths.filter((filePath, index) => exists[index]))
+  .then(foundFilePaths => {
+    if (foundFilePaths.length > 0) {
+        return foundFilePaths[0];
+    }
+    const parentDir = path.resolve(startPath, "../");
```
Comment (Glavin001) [inline src/index.ts]: This line with `parentDir` can be moved to after the `if` statement, since it is not used yet.

File: src/index.ts [hunk 3]
```diff
@@ -53,6 +67,43 @@ export const beautifier: Beautifier = {
...
+  startPath = finishPath,
+  fileNames,
+}: {
+  startPath: string | undefined;
+  finishPath: string | undefined;
+  fileNames: string[];
+}): Promise<string> {
+  // Recursive case: Check & go uuuuppp
+  const filePaths = fileNames.map(fileName => path.join(startPath, fileName));
+  // tslint:disable-next-line
```
Comment (Glavin001) [inline src/index.ts]: What is this disabling?

File: src/index.ts [hunk 4]
```diff
@@ -27,15 +27,29 @@ export const beautifier: Beautifier = {
...
+    .catch((err) => {
+      // tslint:disable-next-line
+      console.log(err);
+      return Promise.resolve({});
+    });
+  },
+  beautify({ text, dependencies, filePath, beautifierConfig }: BeautifierBeautifyData) {
     const phpCsFixer = dependencies.get<ExecutableDependency>("PHP-CS-Fixer");
     const basePath: string = os.tmpdir();
+    const config = beautifierConfig && beautifierConfig.filePath ? `--config=${beautifierConfig.filePath}` : "--rules=@PSR2";
```
Comment (Glavin001) [inline src/index.ts]: I wonder if a better default than `@PSR2` is no configuration?

https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/php-cs-fixer.coffee#L69-L71

File: tsconfig.json [hunk 5]
```diff
@@ -9,7 +9,12 @@
         "target": "es6",
         "declaration": true,
         "moduleResolution": "node",
-        "outDir": "./dist/"
+        "outDir": "./dist/",
+        "baseUrl": ".",
+        "paths": {
+          "@src": ["src"],
+          "@test/*": ["test/*"]
```
Comment (Glavin001) [inline tsconfig.json]: Nice idea!

File: src/index.ts (hunk 3 above)
Comment (szeck87) [reply src/index.ts]: `No need to wrap 'doesFileExist' in another function. Just use it directly.` . https://palantir.github.io/tslint/rules/no-unnecessary-callback-wrapper/

File: src/index.ts (hunk 4 above)
Comment (szeck87) [reply src/index.ts]: If no rules or config is passed, then by default the PSR1 and PSR2 rules are used.

File: src/index.ts [hunk 6]
```diff
@@ -53,6 +66,41 @@ export const beautifier: Beautifier = {
...
+  startPath = finishPath,
+  fileNames,
+}: {
+  startPath: string | undefined;
+  finishPath: string | undefined;
+  fileNames: string[];
+}): Promise<string> {
+  const filePaths = fileNames.map(fileName => path.join(startPath, fileName));
+  // tslint:disable-next-line
+  return Promise.all(filePaths.map(filePath => doesFileExist(filePath)))
```
Comment (Glavin001) [inline src/index.ts]: Change to:

```
return Promise.all(filePaths.map(doesFileExist))
```

File: src/index.ts [hunk 7]
```diff
@@ -27,15 +27,28 @@ export const beautifier: Beautifier = {
...
+      // tslint:disable-next-line
+      console.log(err);
+      return Promise.resolve({});
+    });
+  },
+  beautify({ text, dependencies, filePath, beautifierConfig }: BeautifierBeautifyData) {
     const phpCsFixer = dependencies.get<ExecutableDependency>("PHP-CS-Fixer");
     const basePath: string = os.tmpdir();
+    const config = beautifierConfig && beautifierConfig.filePath ? `--config=${beautifierConfig.filePath}` : "";
+    // tslint:disable-next-line
```
Comment (Glavin001) [inline src/index.ts]: Should be 

```
// tslint:disable-next-line no-console
```

File: src/index.ts [hunk 8]
```diff
@@ -27,15 +27,28 @@ export const beautifier: Beautifier = {
...
     },
   ],
-  beautify({ text, dependencies, filePath }: BeautifierBeautifyData) {
+  resolveConfig: ({ filePath, projectPath }) => {
+    const configFiles: string[] = [".php_cs", ".php_cs.dist"];
+    return findFile({ finishPath: projectPath, startPath: filePath, fileNames: configFiles })
+    .then(configFile => ({ filePath: configFile }))
+    .catch((err) => {
+      // tslint:disable-next-line
+      console.log(err);
```
Comment (Glavin001) [inline src/index.ts]: Should be

```
// tslint:disable-next-line no-console
```

**Pull request merged**